.github/workflows/macstadium-tests.yml

```diff
@@ -175,5 +175,79 @@ jobs:
       - name: Detox iOS e2e tests parallel
         run: |
           ./scripts/run-parallel-e2e.sh
+  # iOS build and e2e tests
+  e2e-new-ios:
+    runs-on: ["self-hosted"]
+    concurrency:
+      group: e2enew-${{ github.workflow }}-${{ github.ref }}
+      cancel-in-progress: true
+    permissions:
+      contents: read
+    timeout-minutes: 90
+    needs: install-deps
+    env:
+      ARTIFACTS_FOLDER: e2e-artifacts
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v4
+
+      - name: Set up github keys
+        run: git config core.sshCommand "ssh -i ~/.ssh/id_ed25519 -F /dev/null"
+
+      - name: Set sanitized branch name
+        id: sanitize
+        run: echo "SANITIZED_REF_NAME=${GITHUB_REF_NAME//\//-}" >> $GITHUB_ENV
+      - name: Download Yarn cache
+        uses: actions/download-artifact@v4
+        with:
+          name: yarn-cache-${{ env.SANITIZED_REF_NAME }}
+          path: .yarn
+      - name: Set up ENV vars & scripts
+        env:
+          CI_SCRIPTS: ${{ secrets.CI_SCRIPTS }}
+        run: |
+          source ~/.zshrc
+          git clone git@github.com:rainbow-me/rainbow-env.git
+          mv rainbow-env/dotenv .env && rm -rf rainbow-env
+          eval $CI_SCRIPTS
+      - name: Install dependencies
+        run: |
+          yarn install && yarn setup
+
+      - name: Version debug
+        run: |
+          npx react-native info
+
+      - name: Install pods
+        run: |
+          rm -rf /Users/administrator/.cocoapods/repos/cocoapods/.git/index.lock
+          yarn install-bundle && yarn install-pods
+
+      - uses: irgaly/xcode-cache@v1
+        with:
+          key: xcode-cache-deriveddata-${{ github.workflow }}-${{ github.sha }}
+          restore-keys: xcode-cache-deriveddata-${{ github.workflow }}-
+
+      - name: Install Maestro
+        run: curl -fsSL "https://get.maestro.mobile.dev" | bash
```

Comment:
Maestro releases seem to be put out quite often, can we select a specific version instead (same for the eventual android flow)